asyncapi/asb/data_types.bal

```diff
@@ -72,21 +101,99 @@ public type AsbError distinct error;
-# Message content type in json format
-public const string JSON = "application/json";
-# Message content type in xml format
-public const string XML = "application/xml";
-# Message content type in byte array format
-public const string BYTE_ARRAY = "application/octet-stream";
+# This record holds the configuration details of a queue in Azure Service Bus
+#
+# + queueName - A string field that holds the name of the queue
+@display {label: "Queue Configurations"}
+public type QueueConfig record {
+    @display {label: "Queue Name"}
+    string queueName;
+};
 
-# Message receive modes.
+# Azure service bus listener configuration.
+public type ListenerConfig record {
+    # The connection string of Azure service bus
+    @display {label: "Connection String", "description": "The connection string of Azure service bus"}
+    string connectionString;
+    # Name or path of the entity (e.g : Queue name, Subscription path)
+    @display {label: "Entity Configuration"}
+    TopicSubsConfig|QueueConfig entityConfig;
+    # Receive mode as PEEKLOCK or RECEIVEANDDELETE (default : PEEKLOCK)
+    @display {label: "Receive Mode", "description": "Receive mode as PEEKLOCK or RECEIVEANDDELETE (default : PEEKLOCK)"}
+    ReceiveMode receiveMode = PEEK_LOCK;
+    # The maximum number (default 1) of concurrent calls (i.e., the maximum number of messages that can be processed at the same time)
+    @display {label: "Number of maximum concurrent calls", "description": 
+    "The maximum number of messages that can be processed at the same time"}
+    int maxConcurrentCalls = 1;
+    # The prefetch count (default 0), representing the number of messages that the listener should receive and buffer
+    @display {label: "Prefetch Count", "description": "The number of messages that the listener should receive and buffer"}
+    int prefetchCount = 0;
+    # The property that controls the maximum duration for which a message lock can be automatically renewed. The lock on 
+    # a message is used to prevent other consumers from processing the same message while it is being processed by the 
+    # current consumer. When the lock is about to expire, it can be automatically renewed if the maxAutoLockRenewDuration 
+    # has not been reached. This property is useful in scenarios where processing a message may take longer than the lock 
+    # duration, to prevent the message from becoming available to other consumers before it is finished processing. 
+    # The value of this property is specified in seconds. Setting Duration#ZERO disables auto-renewal(default)
+    @display {label: "Max Auto Lock Renew Duration in seconds", "description": 
+    "The maximum duration(in seconds) for which a message lock can be automatically renewed"}
+    int maxAutoLockRenewDuration = 0;
+};
+
+# Represents Custom configurations for the ASB connector
+#
+# + logLevel - Enables the connector debug log prints (log4j log levels), default: OFF
+public type CustomConfiguration record {
```

Comment:
What abt renaming `CustomConfiguration to `Options`? 
User cannot input whatever he likes to say it is `custom` right?

Reply:
Then it will be used in Config.toml as follows,
```
[ballerinax.trigger.asb.customConfiguration]
logLevel="OFF"
```
After Change
```
[ballerinax.asb.option]
logLevel="OFF"
```

Reply:
IMO, options is better. Shall we make it plural?

Reply:
+1